src/core/providers/gdal/qgsgdalproviderbase.cpp

```diff
@@ -329,8 +328,25 @@ QVariantMap QgsGdalProviderBase::decodeGdalUri( const QString &uri )
   QStringList openOptions;
 
   QString vsiPrefix = qgsVsiPrefix( path );
-  if ( !path.isEmpty() )
+  QString vsiSuffix;
+  if ( path.startsWith( vsiPrefix, Qt::CaseInsensitive ) )
+  {
     path = path.mid( vsiPrefix.count() );
+    if ( vsiPrefix == QLatin1String( "/vsizip/" ) )
+    {
+      const QRegularExpression vsiRegex( QStringLiteral( "(?:\\.zip|\\.tar|\\.gz|\\.tar\\.gz|\\.tgz)([^|]*)" ) );
```

Comment:
I think the capture group here should be non greedy - which points to a gap in the test coverage

Reply:
@nyalldawson , non greedy not needed, but filling the coverage gap is valid. Done :)